### PR TITLE
docs(torch): add a dedicated section for torch templates

### PR DESCRIPTION
**Pull request by BynaryCobweb**
Includes missing template documentation as well

### CODE REVIEW

**Review by fantes — APPROVED** [review]

File: docs/api.md [hunk 1]
```diff
@@ -746,49 +746,14 @@ self_supervised | string | yes      | ""      | self-supervised mode: "mask" for
...
 offset        | int            | yes      | N/A            | Offset beween start point of sequences with connector `cvsts`, defining the overlap of input series
 forecast_timesteps      | int            | yes      | N/A       | for nbeats model, this gives the length of the forecast
 backcast_timesteps      | int            | yes      | N/A       | for nbeats model, this gives the length of the backcast
 
-
-Model instantiation parameters for recurrent template:
-
-Parameter       | Template  | Type            | Default                      | Description
----------       | --------- | ------          | ---------------------------- | -----------
-layers          | recurrent | array of string | []                           | ["L50","L50"] means 2 layers of LSTMs with hidden size of 50. ["L100","L100", "T", "L300"] means an lstm autoencoder with encoder composed of 2 LSTM layers of hidden size 100 and decoder is one LSTM layer of hidden size 300
```
Comment (beniz) [inline docs/api.md:762]: Attention, this section also applies to timeseries models with Caffe, let's put it back in the appropriate section

Comment (BynaryCobweb) [reply docs/api.md:762]: Shouldn't it be in the Model template section then ? with other caffe template parameters?

Comment (beniz) [reply docs/api.md:762]: As you wish, but let's not forget it.

Comment (BynaryCobweb) [reply docs/api.md:762]: I moved it in a separate section and specified that these parameters are for both backends

**Review by beniz — APPROVED** [review]

**Pull request merged**